fiboa_cli/datasets/ec_si.py

```diff
@@ -0,0 +1,189 @@
+
+# License of the data, either
+# 1. a SPDX license identifier (including "dl-de/by-2-0" / "dl-de/zero-2-0"), or
+LICENSE = "CC-BY-4.0"
+# 2. a STAC Link Object with relation type "license"
+# LICENSE = {"title": "CC-BY-4.0", "href": "https://creativecommons.org/licenses/by/4.0/", "type": "text/html", "rel": "license"}
+
+# Map original column names to fiboa property names
+# You also need to list any column that you may have added in the MIGRATION function (see below).
+COLUMNS = {
+    'geometry': 'geometry', #fiboa core field
+    'ID': 'id', #fiboa core field
+    'AREA': 'area', #fiboa core field
+    'GERK_PID': 'gerk_pid', #fiboa custom field
+    'SIFRA_KMRS': 'crop_type_class', #fiboa custom field
+    'RASTLINA': 'rastlina', #fiboa custom field
+    'CROP_LAT_E': 'crop_lat_e', #fiboa custom field
+    'COLOR': 'color', #fiboa custom field
+    'EC_trans_n': 'EC_trans_n', #fiboa custom field
+    'EC_hcat_n': 'EC_hcat_n', #fiboa custom field
+    'EC_hcat_c': 'EC_hcat_c' #fiboa custom field
+}
+
+# Add columns with constant values.
+# The key is the column name, the value is a constant value that's used for all rows.
+ADD_COLUMNS = {
+    "determination_datetime": "2021-10-06T00:00:00Z"
+}
+
+# A list of implemented extension identifiers
+EXTENSIONS = []
+
+# Functions to migrate data in columns to match the fiboa specification.
+# Example: You have a column area_m in square meters and want to convert
+# to hectares as required for the area field in fiboa.
+# Function signature:
+#   func(column: pd.Series) -> pd.Series
+COLUMN_MIGRATIONS = {}
+
+# Filter columns to only include the ones that are relevant for the collection,
+# e.g. only rows that contain the word "agriculture" but not "forest" in the column "land_cover_type".
+# Lamda function accepts a Pandas Series and returns a Series or a Tuple with a Series and True to inverse the mask.
+COLUMN_FILTERS = {}
+
+# Custom function to migrate the GeoDataFrame if the other options are not sufficient
+# This should be the last resort!
+# Function signature:
+#   func(gdf: gpd.GeoDataFrame) -> gpd.GeoDataFrame
+MIGRATION = None
+
+# Schemas for the fields that are not defined in fiboa
+# Keys must be the values from the COLUMNS dict, not the keys
+MISSING_SCHEMAS = {
+    'required': ['gerk_pid', 'crop_type_class', 'rastlina', 'crop_latin', 'color', 'EC_trans_n','EC_hcat_n', 'EC_hcat_c'], # i.e. non-nullable properties
+    'properties': {
+        'gerk_pid': {
+            'type': 'uint64'
+        },
+        'crop_type_class': {
+            'type': 'string'
+        },
+        'rastlina': {
+            'type': 'string'
+        },
+        'crop_lat_e': {
+            'type': 'string'
+        },
+        'color': {
+            'type': 'string'
+        },
+        'EC_trans_n': {
+            'type': 'string'
+        },
+        'EC_hcat_n': {
+            'type': 'string'
+        },
+        'EC_hcat_c': {
+            'type': 'string'
```

Comment:
The change to string was not the intention. It's a numerical value, I think, but it was just lacking the nullability as the column was required, see my other comments.

Reply:
Now the schema for EC_hcat_c is fully gone. The uint32 in the beginning was fine, you just had to make it optional (as it now is).